src/update_helper.rs

```diff
@@ -0,0 +1,197 @@
+//! # Functions to update timestamps.
+
+use crate::chat::{Chat, ChatId};
+use crate::contact::Contact;
+use crate::context::Context;
+use crate::param::{Param, Params};
+use anyhow::Result;
+
+impl Context {
+    /// Updates a contact's timestamp, if reasonable.
+    /// Returns true if the caller shall update the settings belonging to the scope.
+    /// (if we have a ContactId type at some point, the function should go there)
+    pub(crate) async fn update_contacts_timestamp(
+        &self,
+        contact_id: u32,
+        scope: Param,
+        new_timestamp: i64,
+    ) -> Result<bool> {
+        let mut contact = Contact::load_from_db(self, contact_id).await?;
+        if contact.param.set_timestamp(scope, new_timestamp)? {
+            contact.update_param(self).await?;
+            return Ok(true);
+        }
+        Ok(false)
+    }
+}
+
+impl ChatId {
+    /// Updates a chat id's timestamp on disk, if reasonable.
+    /// Returns true if the caller shall update the settings belonging to the scope.
+    pub(crate) async fn update_timestamp(
+        &self,
+        context: &Context,
+        scope: Param,
+        new_timestamp: i64,
+    ) -> Result<bool> {
+        let mut chat = Chat::load_from_db(context, *self).await?;
+        if chat.param.set_timestamp(scope, new_timestamp)? {
+            chat.update_param(context).await?;
+            return Ok(true);
+        }
+        Ok(false)
+    }
+}
+
+impl Params {
+    /// Updates a param's timestamp in memory, if reasonable.
+    /// Returns true if the caller shall update the settings belonging to the scope.
+    pub(crate) fn set_timestamp(&mut self, scope: Param, new_timestamp: i64) -> Result<bool> {
```

Comment:
Maybe name it `update_timestamp` too?

Reply:
my idea was to be closer to `set_int()`, `set_i64()` etc., however, is it is done only conditionally and also the return value is meaningful, yes, it seems to be better to have the same name here. i will do a commit.